staging/vhost-device-sound/src/virtio_sound.rs

```diff
@@ -342,3 +342,169 @@ pub struct VirtioSoundChmapInfo {
+        let expected_debug = format!(
+            "VirtioSoundJackHeader {{ hdr: {:?}, jack_id: Le32(0) }}",
+            val.hdr
+        );
+        assert_eq!(debug_output, expected_debug);
+
+        let val = VirtioSoundJackInfo::default();
+
+        let debug_output = format!("{:?}", val);
+        let expected_debug = format!(
+            "VirtioSoundJackInfo {{ hdr: {:?}, feature: Le32(0), hda_reg_defconf: Le32(0), \
+             hda_reg_caps: Le32(0), connected: 0, padding: {:?} }}",
+            val.hdr, val.padding
+        );
+        assert_eq!(debug_output, expected_debug);
+
+        let val = VirtioSoundJackRemap::default();
+
+        let debug_output = format!("{:?}", val);
+        let expected_debug = format!(
+            "VirtioSoundJackRemap {{ hdr: {:?}, association: Le32(0), sequence: Le32(0) }}",
+            val.hdr
+        );
+        assert_eq!(debug_output, expected_debug);
+
+        let val = VirtioSoundPcmHeader::default();
+
+        let debug_output = format!("{:?}", val);
+        let expected_debug = format!(
+            "VirtioSoundPcmHeader {{ hdr: {:?}, stream_id: Le32(0) }}",
+            val.hdr
+        );
+        assert_eq!(debug_output, expected_debug);
+
+        let val = VirtioSoundPcmInfo::default();
+
+        let debug_output = format!("{:?}", val);
+        let expected_debug = format!(
+            "VirtioSoundPcmInfo {{ hdr: {:?}, features: Le32(0), formats: Le64(0), rates: \
+             Le64(0), direction: 0, channels_min: 0, channels_max: 0, padding: {:?} }}",
+            val.hdr, val.padding
+        );
+        assert_eq!(debug_output, expected_debug);
+
+        let val = VirtioSndPcmSetParams::default();
+
+        let debug_output = format!("{:?}", val);
+        let expected_debug = format!(
+            "VirtioSndPcmSetParams {{ hdr: {:?}, buffer_bytes: Le32(0), period_bytes: Le32(0), \
+             features: Le32(0), channels: 0, format: 0, rate: 0, padding: 0 }}",
+            val.hdr
+        );
+        assert_eq!(debug_output, expected_debug);
+
+        let val = VirtioSoundPcmXfer::default();
+
+        let debug_output = format!("{:?}", val);
+        let expected_debug = "VirtioSoundPcmXfer { stream_id: Le32(0) }".to_string();
+        assert_eq!(debug_output, expected_debug);
+
+        let val = VirtioSoundPcmStatus::default();
+
+        let debug_output = format!("{:?}", val);
+        let expected_debug =
+            "VirtioSoundPcmStatus { status: Le32(0), latency_bytes: Le32(0) }".to_string();
+        assert_eq!(debug_output, expected_debug);
+
+        let val = VirtioSoundChmapInfo::default();
+
+        let debug_output = format!("{:?}", val);
+        let expected_debug = format!(
+            "VirtioSoundChmapInfo {{ hdr: {:?}, direction: 0, channels: 0, positions: {:?} }}",
+            val.hdr, val.positions
+        );
+        assert_eq!(debug_output, expected_debug);
+    }
+    #[test]
+    fn test_virtiosound_structs_clone() {
```

Comment:
You could probably use `rstest` and [parametrize](https://github.com/la10736/rstest#parametrize) it.

See some examples:
https://github.com/rust-vmm/vhost-device/blob/main/staging/vhost-device-video/src/vhu_video_thread.rs#L690
https://github.com/rust-vmm/vhost-device/blob/main/staging/vhost-device-sound/src/main.rs#L68

Reply:
Good point!